xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -178,87 +217,107 @@ static ITestCase Deserialize(LoggerHelper logger, ITestFrameworkExecutor executo
 
                 using (AssemblyHelper.SubscribeResolve())
                 {
-                    var reporterMessageHandler = GetRunnerReporter(sources).CreateMessageHandler(new VisualStudioRunnerLogger(logger));
-
                     foreach (var assemblyFileNameCanBeWithoutAbsolutePath in sources)
                     {
-                        var assemblyFileName = assemblyFileNameCanBeWithoutAbsolutePath;
-#if !WINDOWS_UAP
-                        assemblyFileName = Path.GetFullPath(assemblyFileNameCanBeWithoutAbsolutePath);
-#endif
-                        var assembly = new XunitProjectAssembly { AssemblyFilename = assemblyFileName };
+                        var assemblyFileName = GetAssemblyFileName(assemblyFileNameCanBeWithoutAbsolutePath);
                         var configuration = LoadConfiguration(assemblyFileName);
                         var fileName = Path.GetFileNameWithoutExtension(assemblyFileName);
                         var shadowCopy = configuration.ShadowCopyOrDefault;
 
-                        try
+                        var diagnosticSink = new DiagnosticMessageSink(logger, fileName, configuration.DiagnosticMessagesOrDefault);
+                        using (var framework = new XunitFrontController(AppDomainDefaultBehavior,
```

Comment:
This XunitFrontController is only created for `Discovery`. We provide an existing controller to `DiscoverTestsInSource` in case of Discovery during Run All.